python/tirex_tracker/archive_utils.py

```diff
@@ -55,13 +55,25 @@ def _create_notebook_zip_archive(
     with redirect_stdout(None):
         ipython.run_line_magic("save", f"-f {script_file_path} 1-9999")
         ipython.run_line_magic("notebook", str(notebook_file_path))
-    return _create_zip_archive(
+    ret = _create_zip_archive(
         metadata_directory_path=metadata_directory_path,
         base_directory_path=metadata_directory_path,
         script_file_path=script_file_path,
         notebook_file_path=notebook_file_path,
     )
 
+    script_file_path.unlink()
+    notebook_file_path.unlink()
+
+    return ret
+
+
+def git_repo_or_none(script_file_path: Path) -> Optional[Repo]:
+    try:
+        return Repo(script_file_path, search_parent_directories=True)
+    except InvalidGitRepositoryError:
+        return None
+
```

Comment:
Why move this simple check out of the `create_git_zip_archive()` function?
To me, having the "speaking" exception name in the main logic is more interpretable than the generic `git_repo_or_none()`, e.g., the information is lost when it will be `None` and when not.

Reply:
Because I needed this function at a second place as well. It is opinionated if the speaking exception is prefered or not, for me it was just the other way around.